accepted/00097-recurring-actions.md

```diff
@@ -0,0 +1,193 @@
+
+Further research on how to make the above work (if desired) should be done in it's own dedicated RFC due to it's expected complexity.
+
+Once Action Chains operate in before mentioned way they can be implemented for recurring execution in the same way:
+
+1) Create a `RecurringActionChainType` entity.
+2) Add an `action_chain_id` column to the `suseRecurringActionType` that would reference the `ActionChain` object.
+3) Add backend support for the ActionChainType.
+4) Adapt the `RecurringActionApplyJob` `Taskomatic` job to support the type.
+5) Allow CRUD operations to be performed on the frontend side.
+
+Recurring Action Chains could be accessed from the main menu `Schedule` -> ` Action Chains` -> `Recurring (Action Chains)`.
+
+### OpenSCAP scans
+
+To make openSCAP scans work properly with recurring actions we need to be able to create persistent openSCAP scan profiles that would be reused on every action execution. Currently openSCAP scans can be scheduled using a set of parameters and rescheduled only from the results of a previously performed scan.
+
+These profiles should also be able to apply to entire system groups and orgs to make recurring schedules on group/org level work.
+
+The `suseRecurringActionType` table would then only need to include an id referencing the openSCAP profile entity.
+
+The rest of the implementation would work similar to other action types.
+
+### Ansible Playbook execution
+
+TODO
+
+## Traditional Clients
+
+Even though traditional clients technically support openSCAP scans as well as action chains. I propose not to implement any recurring functionality for them. We want to stop supporting them and incentivize users to migrate to salt minions.
+
+## Maintenance Windows
+
+For recurring action scheduling to work properly the schedules needs to be aligned with existing maintenance windows on the clients. Otherwise the action is not going to be scheduled with the exception of actions allowed when the system is not in maintenance mode (openSCAP scans).
+
+## XMLRPC / HTTP API
+
+We already have a `RecurringActionHandler` that can be expanded to support other action types.
+
+I see two possibilities two implement new functions:
+
+1) For each type implement individiual CRUD function that states the action type in its name. E.g. `RecurringActionHandler.createCustomState(...)`. This is essentially just a wrapper that calls the generic `create` function with the action type as a parameter.
+
+2) Have one CRUD function for all the types and pass the action type as an argument.
+
+Option 1) could be more intuitive to use and its function would be clear just by reading its name, but add a lot of extra API endpoints that all call the same underlying base function, so option 2) would be less overhead.
+
+In the end I don't have a strong opinion which approach to take.
```

Comment:
I would say that we can follow the approach that would be simpler and less error prom for users, which looks like option 1. But I also don't have a strong opinion on this